src/connection.ts

```diff
@@ -234,6 +248,10 @@ export class Connection {
     return res
   }
 
+  public async askForCredit(params: CreditRequestParams): Promise<void> {
+    return await this.send(new CreditRequest({ ...params }))
+  }
```

Comment:
I don't think we have to expose this functionality to the final user, I think we have to embed it in the consumer, check the .NET client to see an example

Reply:
Ok, i saw that the on the .NET client the credit is used on the RawConsumer, should we replicate this the same way in our consumer? Can I use the RawConsumer as a reference?